crates/storage/storage-api/src/trie.rs

```diff
@@ -41,6 +42,33 @@ pub trait StateRootProvider: Send + Sync {
     ) -> ProviderResult<(B256, TrieUpdates)>;
 }
 
+/// A trait that is used to compute the state root of the latest state stored in the database.
```

Comment:
wondering if `ProviderError` needs a new variant `ProviderError::Custom(String)` for this to be fully extensible?

Reply:
Yes, this would be useful. Shall I go ahead and include it in this PR?

Reply:
an idea is to change the base in that case? so prs are stacked on top of each other, could speed up review process but may slow you down

Reply:
The issue is that the PR's would then target my private fork if I changed the base to the preceding PR.